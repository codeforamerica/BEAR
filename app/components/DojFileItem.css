.fileName {
  display: flex;
  align-items: center;
}

.fileRemove {
  padding-left: 5px;
  cursor: pointer;
}
