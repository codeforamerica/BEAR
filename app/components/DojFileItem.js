// @flow
import React, { Component } from 'react';
import path from 'path';
import styles from './DojFileItem.css';

type Props = {
  filePath: string,
  onFileRemove: string => void
};

export default class DojFileItem extends Component<Props> {
  formatFileName = () => {
    const { filePath } = this.props;
    return filePath.split(path.sep).pop();
  };

  handleFileRemove = () => {
    const { onFileRemove } = this.props;
    onFileRemove('');
  };

  render() {
    return (
      <div className="doj-file">
        <p className={styles.fileName}>
          File imported: {this.formatFileName()}
          {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/interactive-supports-focus */}
          <i
            className={`${styles.fileRemove} icon-close`}
            onClick={this.handleFileRemove}
            role="button"
          />
        </p>
      </div>
    );
  }
}
