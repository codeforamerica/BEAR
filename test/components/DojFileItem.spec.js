import sinon from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import DojFileItem from '../../app/components/DojFileItem';

Enzyme.configure({ adapter: new Adapter() });
const sandbox = sinon.createSandbox();

function setup(filePath) {
  const onFileRemoveSpy = sandbox.spy();
  const component = shallow(
    <DojFileItem filePath={filePath} onFileRemove={onFileRemoveSpy} />
  );
  return {
    component,
    onFileRemoveSpy
  };
}

afterEach(() => {
  sandbox.restore();
});

describe('DojFileItem component', () => {
  it('should display the file name from path', () => {
    const { component } = setup('path/to/file.dat');
    const fileName = component.find('.fileName');
    expect(fileName.text()).toEqual('File imported: file.dat');
  });

  it('should call onFileRemove with the clicked path when the close icon is clicked', () => {
    const { component, onFileRemoveSpy } = setup('path/to/file.dat');
    component.find('.icon-close').simulate('click');
    expect(onFileRemoveSpy.called).toBe(true);
    const { args } = onFileRemoveSpy.getCall(0);
    expect(args[0]).toEqual('path/to/file.dat');
  });

  it('should match exact snapshot', () => {
    const { component } = setup('path/to/file');
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
