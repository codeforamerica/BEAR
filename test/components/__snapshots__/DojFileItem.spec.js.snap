// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DojFileItem component should match exact snapshot 1`] = `
<div
  className="doj-file"
>
  <p
    className="fileName"
  >
    File imported: 
    file
    <i
      className="fileRemove icon-close"
      onClick={[Function]}
      role="button"
    />
  </p>
</div>
`;
